docs/OnTrack/Front End Migration/Research & Findings/Testing Decision.md

```diff
@@ -0,0 +1,95 @@
+## What is the current state of testing?
+In this issue, we will discuss the current state of testing and what tools are we proposing.
+While there are some existing unit tests for components written in TypeScript, there are no existing
+unit tests for the CoffeeScript components. Existing tests use Karma test runner integrated with
+Jasmine. Currently, the ng test command (Runs unit tests in a project) and npm install are not
+working due to dependencies issues, which are getting fixed.
+
+## Comparing different alternatives
+
+## Cypress
+
+## Pros
+
+-Selenium involves implementing browser drivers for the script to communicate with the web elements

```

Comment:
@Ishrat420 I know this is also mentioned in [Unit Testing #7](https://github.com/thoth-tech/doubtfire-web/issues/7#issue-1206279940) but just to be clear, is the implication here that because Selenium implements browser drivers, this is a pro for Cypress because Cypress does not and is therefore faster/better/more-efficient etc? I wonder if this could be written more explicitly to make it clearer?

(And great work on all of this. It's excellent 🥇 )

Reply:
Yes you are right. Browser drivers help to run selenium scripts on the browser. They act as a bridge that interprets the selenium commands to execute it in the browser. There are separate browser drivers for each browser e.g ChromeDriver, FirefoxDriver etc. Cypress does not utilize WebDriver for testing. If your language can someway be transpiled to JS, it can use DOM events to send the click command to the button or such. This results in a much faster execution of test results